pkg/eks/eks.go

```diff
@@ -69,72 +67,11 @@ func (c *ClusterProvider) RefreshClusterStatus(spec *api.ClusterConfig) error {
 	}
 }
 
-// SupportsManagedNodes reports whether an existing cluster supports Managed Nodes
-// The minimum required control plane version and platform version are 1.14 and eks.3 respectively
-func (c *ClusterProvider) SupportsManagedNodes(clusterConfig *api.ClusterConfig) (bool, error) {
-	if err := c.RefreshClusterStatusIfStale(clusterConfig); err != nil {
-		return false, err
-	}
-
-	return ClusterSupportsManagedNodes(c.Status.ClusterInfo.Cluster)
-}
-
 // isNonEKSCluster returns true if the cluster is external
 func isNonEKSCluster(cluster *awseks.Cluster) bool {
 	return cluster.ConnectorConfig != nil
 }
 
-// ClusterSupportsManagedNodes reports whether the EKS cluster supports managed nodes
-func ClusterSupportsManagedNodes(cluster *awseks.Cluster) (bool, error) {
-	if cluster.PlatformVersion == nil {
-		logger.Warning("could not find cluster's platform version")
-		return false, nil
-	}
-	version, err := PlatformVersion(*cluster.PlatformVersion)
-	if err != nil {
-		return false, err
-	}
-	minSupportedVersion := 3
-	return version >= minSupportedVersion, nil
```

Comment:
this was the exact fix for the linked issue, the others changes are just more redundant checks I noticed